cmake/winml.cmake

```diff
@@ -472,6 +474,7 @@ target_link_libraries(winml_dll PRIVATE winml_lib_api)
 target_link_libraries(winml_dll PRIVATE winml_lib_image)
 target_link_libraries(winml_dll PRIVATE winml_lib_telemetry)
 target_link_libraries(winml_dll PRIVATE onecoreuap_apiset.lib)
+target_link_libraries(winml_dll PRIVATE delayimp.lib)
```

Comment:
What does this do?

Reply:
Linking was failing due to missing a implementation of `__delayLoadHelper2`. [According to the docs](https://docs.microsoft.com/en-us/cpp/build/reference/specifying-dlls-to-delay-load?view=vs-2019), `delayimp.lib` must be linked when delay loading libraries or a custom implementation of `__delayLoadHelper2` must be provided: `If you do not plan to use your own version of a helper function, you must also link your program with delayimp.lib (for desktop applications) or dloadhelper.lib (for store apps)`

I don't know why it worked with msbuild, maybe it added delayimp.lib automatically?

Reply:
Linking was failing due to missing a implementation of __delayLoadHelper2. According to the docs, delayimp.lib must be linked when delay loading libraries or a custom implementation of __delayLoadHelper2 must be provided: If you do not plan to use your own version of a helper function, you must also link your program with delayimp.lib (for desktop applications) or dloadhelper.lib (for store apps)

I don't know why it worked with msbuild, maybe it added delayimp.lib automatically?

Reply:
can you make sure this doesn't add dependencies on desktop-only binaries? We need to build for onecore-based SKUs as well.

Reply:
I'll run msbuild and look at its linker command line to find out why it works there and not with ninja.

If I find out that delayimp is really needed, how can I test in other platforms to make sure I didn't berak non-desktop builds?